codes/classical/q-ary_digits/quasi_cyclic.yml

```diff
@@ -10,8 +10,10 @@ logical: q-ary_digits
 name: 'Quasi-cyclic code'
 introduced: '\cite{doi:10.1109/TIT.1967.1053974}'
 
-description: 'Stub.'
+description: 'Given a finite field \(\mathbb{F}_q\), a quasi-cyclic of index \(k\) code over this field is a code \(C\) for which any codeword \((c_1, \dots, c_n)\in C\)' shifted by \(k\) units is in \(C\), and \(k\) is the minimal number of indices for which this holds.\cite{https://arxiv.org/abs/2007.16029}. Equivalently, a quasi-cyclic code is an \(S\)-submodule of \(\mathbb{F}_q^{n}\), where \(S = \frac{\mathbb{F}_q}{\langle x^n-1\rangle\). A special case of quasi-cyclic codes are the cyclic codes, which have index 1. 
```

Comment:
Formulate for a general alphabet R like the entry code_id:skew_cyclic. Start a new paragraph and write out the submodule part, specializing to the alphabet \(R\) being a field.